Form/EventListener/ResizePolyFormListener.php

```diff
@@ -84,7 +85,12 @@ public function __construct(array $prototypes, array $options = array(), $allowA
                 $defaultType = $type;
             }
 
-            $typeKey = $type instanceof FormTypeInterface ? $type->getName() : $type;
+            $typeKey = $type;
+
+            if ($type instanceof FormTypeInterface) {
```

Comment:
:+1: 

I suggest triggering this only, when using Symfony 2.8+
You can use `LegacyFormUtil::isFullClassNameRequired()` to check:
```php
if (LegacyFormUtil::isFullClassNameRequired()) {
    @trigger_error(sprintf('Passing type instances to PolyCollection is deprecated since version 1.0.5 and will not be supported in 2.0. Use the fully-qualified type class name instead (%s).', get_class($type)), E_USER_DEPRECATED);
}
```

So your deprecation is in sync to the same behavior of the used Symfony version.